cmd/snap-repair/runner_test.go

```diff
@@ -432,14 +460,103 @@ func (s *runnerSuite) TestLoadStateInitState(c *C) {
 	// sanity
 	c.Check(osutil.IsDirectory(dirs.SnapRepairDir), Equals, false)
 	c.Check(osutil.FileExists(dirs.SnapRepairStateFile), Equals, false)
+	// setup realistic seed/assertions
+	seedAssertsDir := filepath.Join(dirs.SnapSeedDir, "assertions")
+	err := os.MkdirAll(seedAssertsDir, 0775)
+	c.Assert(err, IsNil)
+	err = ioutil.WriteFile(filepath.Join(seedAssertsDir, "store.account-key"), asserts.Encode(s.storeSigning.StoreAccountKey("")), 0644)
+	c.Assert(err, IsNil)
+	err = ioutil.WriteFile(filepath.Join(seedAssertsDir, "brand.account"), asserts.Encode(s.brandAcct), 0644)
+	c.Assert(err, IsNil)
+	err = ioutil.WriteFile(filepath.Join(seedAssertsDir, "brand.account-key"), asserts.Encode(s.brandAcctKey), 0644)
+	c.Assert(err, IsNil)
+	modelAs, err := s.brandSigning.Sign(asserts.ModelType, map[string]interface{}{
+		"series":       "16",
+		"brand-id":     "my-brand",
+		"model":        "my-model-2",
+		"architecture": "armhf",
+		"gadget":       "gadget",
+		"kernel":       "kernel",
+		"timestamp":    time.Now().UTC().Format(time.RFC3339),
+	}, nil, "")
+	c.Assert(err, IsNil)
+	err = ioutil.WriteFile(filepath.Join(seedAssertsDir, "model"), asserts.Encode(modelAs), 0644)
+	c.Assert(err, IsNil)
+
+	r := sysdb.InjectTrusted(s.storeSigning.Trusted)
+	defer r()
+
 	runner := repair.NewRunner()
-	err := runner.LoadState()
+	err = runner.LoadState()
 	c.Assert(err, IsNil)
 	c.Check(osutil.FileExists(dirs.SnapRepairStateFile), Equals, true)
-	// TODO: init state will do more later
+
 	brand, model := runner.BrandModel()
-	c.Check(brand, Equals, "")
-	c.Check(model, Equals, "")
+	c.Check(brand, Equals, "my-brand")
+	c.Check(model, Equals, "my-model-2")
+}
+
+func (s *runnerSuite) TestLoadStateInitDeviceInfoFail(c *C) {
+	// sanity
+	c.Check(osutil.IsDirectory(dirs.SnapRepairDir), Equals, false)
+	c.Check(osutil.FileExists(dirs.SnapRepairStateFile), Equals, false)
+	// setup realistic seed/assertions
+	seedAssertsDir := filepath.Join(dirs.SnapSeedDir, "assertions")
+	err := os.MkdirAll(seedAssertsDir, 0775)
+	c.Assert(err, IsNil)
+
+	modelAs, err := s.brandSigning.Sign(asserts.ModelType, map[string]interface{}{
```

Comment:
Parts of this look similar to the setup a few lines above, maybe extract a small helper?